ts/examples/vercel/src/tool-router-ai.ts

```diff
@@ -0,0 +1,84 @@
+import { openai } from '@ai-sdk/openai';
+import { createMCPClient } from "@ai-sdk/mcp"
+import { Composio } from '@composio/core';
+import { VercelProvider } from '@composio/vercel';
+import { stepCountIs, streamText } from 'ai';
+
+// 1. Initialize Composio.
+const composio = new Composio({
+  provider: new VercelProvider(),
+});
+
+// 2. Create an MCP session
+console.log('🔄 Creating toolrouter session...');
+const session = await composio.create('jkomyno', {
+  toolkits: ['gmail'], 
+  manageConnections: true,
+  tools: {
+    'gmail': {
+      enable: ['GMAIL_FETCH_EMAILS'],
+    }
+  }
+});
+
+const { mcp, sessionId } = session;
+
+console.log(JSON.stringify(mcp, null, 2));
+console.log(`✅ Toolrouter session created: ${sessionId}`);
+
+// 3. Create an MCP client
+console.log(`🔄 Connecting to MCP Server: ${mcp.url}`);
+const mcpClient = await createMCPClient({
+  transport: {
+    type: 'http',
+    url: mcp.url,
+    headers: mcp.headers
+  }
+});
+
+// 4. Retrieve tools.
+console.log(`🔄 Retrieving tools...`);
+const tools = await session.tools();
```

Comment:
While using MCP tools should be extracted from the MCP client and not from the session. Only during native tool use we should extract it from the session.